plugins/modules/cloud/google/gcp_storage_aws_transfer_job.py

```diff
@@ -0,0 +1,314 @@
+  loop: "{{ gcp_create_transfer_jobs }}"
+
+
+'''
+
+RETURN = '''
+message:
+    description: Message indicating whether the job was created or deleted.
+    type: str
+    returned: always
+'''
+
+
+try:
+    import googleapiclient.discovery
+    from google.oauth2 import service_account
+    HAS_GOOGLE = True
+except ImportError as e:
+    HAS_GOOGLE = False
+
+import datetime
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def _validate_params(params):
+    """
+    Validate params.
+
+    This function calls _validated_date to verify
+    date time parameters.
+
+    :param params: Ansible dictionary containing configuration.
+    :type  params: ``dict``
+
+    :return: True or raises ValueError
+    :rtype: ``bool`` or `class:ValueError`
+    """
+    fields = [
+        {'name': 'scheduled_start_date_utc', 'type': str, 'required': True},
+        {'name': 'scheduled_end_date_utc', 'type': str, 'required': True},
+        {'name': 'scheduled_start_time_utc', 'type': str, 'required': True},
+    ]
+
+    date_format = '%Y/%m/%d'
+    time_format = '%H:%M'
+
+    try:
+        if 'scheduled_start_date_utc' in params and params['scheduled_start_date_utc'] is not None:
+                _validated_date(params['scheduled_start_date_utc'], date_format)
+        if 'scheduled_end_date_utc' in params and params['scheduled_end_date_utc'] is not None:
+                _validated_date(params['scheduled_end_date_utc'], date_format)
+        if 'scheduled_start_time_utc' in params and params['scheduled_start_time_utc'] is not None:
+                _validated_date(params['scheduled_start_time_utc'], time_format)
+    except Exception:
+        raise
+
+    return (True, '')
+
+
+def _validated_date(param, format):
+    try:
+        value = datetime.datetime.strptime(param, format)
+    except:
+        raise ValueError("Incorrect format. Date and Times must be formatted as YYYY/MM/DD and HH:SS")
+
+
+def run_module():
+
+    module_args = dict(
+        project_id=dict(type='str', required=True),
+        scheduled_start_date_utc=dict(type='str', required=True),
+        scheduled_end_date_utc=dict(type='str', required=True),
+        scheduled_start_time_utc=dict(type='str', required=True),
+        gcp_storage_bucket=dict(type='str', required=True),
+        service_account_file=dict(type='str', required=True),
+        aws_s3_bucket=dict(type='str', required=True),
+        aws_s3_bucket_prefix=dict(type='str', required=False),
+        aws_access_key=dict(type='str', required=True),
+        aws_secret_key=dict(type='str', required=True),
+        description=dict(type='str', required=True),
+        state=dict(type='str', required=True)
+    )
+
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        module.exit_json(**result)
```

Comment:
```suggestion
    if module.check_mode:
        module.exit_json(**result)
```
what does this check_mode implementation give users?
wouldn't it be more fair not to support check_mode if there are no any checks?